src/lib/operators/join_hash.cpp

```diff
@@ -617,8 +627,49 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
         _mode(mode),
         _column_ids(column_ids),
         _predicate_condition(predicate_condition),
-        _inputs_swapped(inputs_swapped),
-        _radix_bits(radix_bits) {}
+        _inputs_swapped(inputs_swapped) {
+          /*
+            Setting number of bits for radix clustering:
+            The number of bits is used to create probe partitions with a size that can
+            be expected to fit into the L2 cache.
+            This should incorporate hardware knowledge, once available in Hyrise.
+            As of now, we assume a L2 cache size of 256 KB.
+            We estimate the size the following way:
+              - we assume each key appears once (that is an overestimation space-wise, but we
+              aim rather for a hash map that is slightly smaller than L2 than slightly larger)
+              - each entry in the hash map is a data structure holding the actual value
+              and the RowID
+          */
+          const auto build_relation_size = inputs_swapped ? _right->get_output()->row_count() : _left->get_output()->row_count();
```

Comment:
Please verify, but I think that `inputs_swapped` means that the inputs already *were* swapped, not that you have to do it again.

Reply:
Yep, right. Done.